src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -0,0 +1,359 @@
+                                name="Settings_Preferences"
+                                component={SettingsPreferencesPage}
+                                options={{
+                                    ...defaultSubRouteOptions,
+                                    title: 'Settings',
+                                }}
+                            />
+                            <SettingsModalStack.Screen
+                                name="Settings_Password"
+                                component={SettingsPasswordPage}
+                                options={{
+                                    ...defaultSubRouteOptions,
+                                    title: 'Settings',
+                                }}
+                            />
+                            <SettingsModalStack.Screen
+                                name="Settings_Payments"
+                                component={SettingsPaymentsPage}
+                                options={{
+                                    ...defaultSubRouteOptions,
+                                    title: 'Settings',
+                                }}
+                            />
+                        </SettingsModalStack.Navigator>
+                    )}
+                </RootStack.Screen>
+                <RootStack.Screen
+                    name="NewChat"
+                    options={modalScreenOptions}
+                >
+                    {() => (
+                        <NewChatModalStack.Navigator
+                            path="/new/chat"
+                        >
+                            <NewChatModalStack.Screen
+                                name="NewChat_Root"
+                                component={NewChatPage}
+                                options={{
+                                    ...defaultSubRouteOptions,
+                                    title: 'New Chat',
+                                }}
+                            />
+                        </NewChatModalStack.Navigator>
+                    )}
+                </RootStack.Screen>
+                <RootStack.Screen
+                    name="NewGroup"
+                    options={modalScreenOptions}
+                >
+                    {() => (
+                        <NewGroupModalStack.Navigator
+                            path="/new/group"
+                        >
+                            <NewGroupModalStack.Screen
+                                name="NewGroup_Root"
+                                component={NewGroupPage}
+                                options={{
+                                    ...defaultSubRouteOptions,
+                                    title: 'New Group',
+                                }}
+                            />
+                        </NewGroupModalStack.Navigator>
+                    )}
+                </RootStack.Screen>
+                <RootStack.Screen
```

Comment:
Each modal needs it's own navigator since we are
- pushing a modal screen onto the main stack (see `mode="modal"`) [docs](https://reactnavigation.org/docs/modal/#creating-a-modal-stack)
- we're also using a custom navigator on web
- having a navigator inside each modal screen allows us to navigate to the various sub screens while remaining inside the modal

There might be some better configuration here. But I think this works for now. Will add some comments to the code to better explain the current reasoning.

Reply:
This is the right way to set up modals, screens should be grouped based on the animation/navigation functionality desired which means they screens can also be defined in multiple stacks.

Having them all live in the root would cause each screen to have the "modal" animation when navigating through the screens and it would be easy to dismiss all the screens at once.

This setup makes it so navigating to the screen opens the modal ↑ animation, navigating between screens in the modal stack have ←→ animations and dismissing the modal from any screen would close the modal ↓